RFC-008-ERC20.md

```diff
@@ -0,0 +1,97 @@
+# The ERC20 Asset
+
+- RFC-Number: 008
+- Status: Draft
+- Discussion-issue: -
+- Created on: 2019-03-12
+
+## Table of Contents
+<!-- markdown-toc start - Don't edit this section. Run M-x markdown-toc-refresh-toc -->
+- [Description](#description)
+- [The ERC20 Asset](#the-erc20-asset-1)
+    - [`quantity`](#quantity)
+    - [`address`](#address)
+- [Registry extension](#registry-extension)
+- [Examples](#examples)
+<!-- markdown-toc end -->
+
+
+## Description
+
+This RFC defines how Ethereum based [ERC20](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-20.md) tokens should be described in Asset type headers.
+
+The Asset type header was introduced in [RFC002](./RFC-002-SWAP.md) to describe assets being exchanged in a COMIT SWAP protocol.
+
+## The ERC20 Asset
+
+The ERC20 token standard was introduced in [EIP20](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-20.md).
+Token contracts are record how much of the token is owned by particular addresses.
+
+Each ERC20 token contract defines its name.
+Just as with the native Ether asset, the nominal token quantity is measured in a smaller denomination which we will call *token wei*.
+The relationship between the nominal asset and the token wei is defined per contract.
+For the purposes of this RFC we ignore nominal amounts completely and simply refer to quantities of token wei.
+
+To describe ERC20 asset in an Asset type header specify `erc20` as the value along with the following parameters:
+
+### `quantity`
+
+The `quantity` paramter describes the quantity of token wei the asset represents.
+The `quantity` parameter is mandatory.
+Its value MUST be a `u256` (note that in the JSON encoding a `u256` is encoded as decimal string like `"1000000000000000000"`).
+
+### `address`
```

Comment:
:x: I preferred the parameter name `token_contract`. Maybe we could also go for `token_identity`?

Reply:
`address` sounds so generic ...

Reply:
Agreed. This was just me specifying how it's already implemented.